controllers/config.go

```diff
@@ -0,0 +1,57 @@
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	configv1 "github.com/openshift/api/config/v1"
+)
+
+// imagesReference allows build systems to inject imagesReference for CCCMO components
+type imagesReference struct {
+	CloudControllerManagerAWS       string `json:"cloudControllerManagerAWS"`
+	CloudControllerManagerOpenStack string `json:"cloudControllerManagerOpenStack"`
+}
+
+// operatorConfig contains configuration values for templating resources
+type operatorConfig struct {
+	ManagedNamespace string
+	ControllerImage  string
+}
+
+func getProviderFromInfrastructure(infra *configv1.Infrastructure) (configv1.PlatformType, error) {
+	if infra == nil || infra.Status.PlatformStatus == nil {
+		return "", fmt.Errorf("platform status is not pupulated on infrastructure")
+	}
+	if infra.Status.PlatformStatus.Type == "" {
+		return "", fmt.Errorf("no platform provider found on infrastructure")
+	}
+
+	return infra.Status.PlatformStatus.Type, nil
+}
+
+func getImagesFromJSONFile(filePath string) (imagesReference, error) {
+	data, err := ioutil.ReadFile(filepath.Clean(filePath))
```

Comment:
are these file paths loaded as volume mounts in the container?

Reply:
Yes, they are loaded from a mounted config map introduced in https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/31